Kira_iris_regression.Rmd

```diff
@@ -0,0 +1,82 @@
+---
+title: "Kira's Regression"
+author: "Kira Webster"
+date: "2020-01-22"
+output: html_document
+---
+
+Starting with the iris data like usual, let's get our dimensions and names.
+```{r}
+iris	# Fisherʻs iris dataset
+dim(iris)	# dimensions
+names(iris)
+```
+
+## Renaming iris with more convenient names
+## Accessing vectors within iris, saving as separate objects
+Next, we will rename all of our iris columns with easier names. Then, we define our variables, species, x, and y.
+```{r}
+names(iris) <- c("sepalL", "sepalW",  "petalL", "petalW", "species")
+species <- iris$species   
+x <- iris$petalL
+y <- iris$petalW
+```
+
+## Illustrating regression by graphing 
+These are examples of how to set your intercept and slope. Now we can plot our x and y plot, and create a line of best fit.
+```{r}
+fit <- lm( y ~ x )  #fitting the linear model
+b <- coef( fit ) #grabbing the coefficients
+int <- b[1] #intercept
+slope <- b[2] #slope
+
+plot(x, y) # plotting the data points
+abline(fit) # drawing the regression line on the plot	#look up help for ?abline and figure out how to use		#abline by specify slope and intercept.
+yhat <- slope*x + int #the fitted values
+points( x, yhat, pch=3)  #adding the fitted points
+
+i=150	#If we want to draw line segments between the 150th point and the fitted regression value for it:
+segments( x[i], y[i], x[i], yhat[i], col="red" )
+segments( x, y, x, yhat, col="red", lty=2 )	#R is "vectorized": it will work on vectors all at once:
+
+sse <- (yhat-y)		#Using previously defined formulas for y and yhat, we can determine the SSE.
+SSE <- sum((sse))^2
+
+text(2,2.5, paste("Slope:", slope)) 	#This section is just to add text on top of the graph. The first two numbers are the x and y coordinates that the text will be placed at, then the part in the quotation marks is the title of the text, and the last bit is the variable that will be displayed next to the title.
```

Comment:
wondering if you could condense (this is also a note to myself for my code)?

Reply:
Probably, wanted it descriptive in case I repeat it, but it is a lot.